src/viam/sdk/services/motion/motion.hpp

```diff
@@ -247,6 +366,63 @@ class Motion : public Service {
         const std::vector<WorldState::transform>& supplemental_transforms,
         const AttributeMap& extra) = 0;
 
+    /// @brief Stop a currently executing motion plan.
+    /// @param component_name the component of the currently executing plan to stop.
+    inline void stop_plan(const Name& component_name) {
+        return stop_plan(component_name, {});
+    }
+
+    /// @brief Stop a currently executing motion plan.
+    /// @param component_name the component of the currently executing plan to stop.
+    /// @param extra Any additional arguments to the method.
+    virtual void stop_plan(const Name& component_name, const AttributeMap& extra) = 0;
+
+    /// @brief Returns the plan and state history of the most recent execution to move a component.
+    /// Returns a result if the last execution is still executing, or changed state within the last
+    /// 24 hours without an intervening robot reinitialization.
+    /// @param component_name The name of the component which the MoveOnGlobe request asked to move.
+    /// @param last_plan_only If true, causes only the last plan for the component/execution to be
+    /// returned. Defaults to false.
+    /// @param execution_id Optional execution id of a previous plan, if you want to know about it.
+    inline std::pair<plan_with_status, std::vector<plan_with_status>> get_plan(
+        Name component_name,
+        bool last_plan_only = false,
+        boost::optional<std::string> execution_id = {}) {
+        return get_plan(component_name, {}, last_plan_only, execution_id);
+    }
+
+    /// @brief Returns the plan and state history of the most recent execution to move a component.
+    /// Returns a result if the last execution is still executing, or changed state within the last
+    /// 24 hours without an intervening robot reinitialization.
+    /// @param component_name The name of the component which the MoveOnGlobe request asked to move.
+    /// @param last_plan_only If true, causes only the last plan for the component/execution to be
+    /// returned.
+    /// @param execution_id Optional execution id of a previous plan, if you want to know about it.
+    /// @param extra Any additional arguments to the method.
+    virtual std::pair<plan_with_status, std::vector<plan_with_status>> get_plan(
+        const Name& component_name,
+        const AttributeMap& extra,
+        bool last_plan_only = false,
+        boost::optional<std::string> execution_id = {}) = 0;
+
+    /// @brief Returns the status of plans created by MoveOnGlobe requests.
+    /// Includes statuses of plans that are executing, or are part of an executing that changed
+    /// its state within the last 24 hours.
+    /// @param only_active_plans If true, filters returned plans to only those currently active.
+    /// Defaults to false.
+    inline std::vector<plan_status_with_id> list_plan_statuses(bool only_active_plans = false) {
```

Comment:
Ditto here: I'd make this two methods: `list_plan_statusus` and `list_active_plan_statuses`.